src/nvim/terminal.c

```diff
@@ -726,6 +726,9 @@ static int terminal_execute(VimState *state, int key)
     FALLTHROUGH;
 
   default:
+    if (key == Ctrl_C) {
+      got_int = false;
```

Comment:
how is anyone going to know the purpose of this in the future? git spelunking which leads to... a gigantic github discussion... is costly. We can easily compress the conclusion into a minimal comment.

```suggestion
      // terminal_enter unconditionally sets `mapped_ctrl_c` to avoid got_int. 8eeda7169aa4
      // But got_int may be set elsewhere, e.g. by interrupt(). So ensure that it is cleared.
      got_int = false;
```

Reply:
After this change, do we even need the `mapped_ctrl_c |= ...` line?

Reply:
> After this change, do we even need the `mapped_ctrl_c |=` ... line?

For the purpose of `got_int` in terminals, it woild be safe to  assume it is no longer required.

That said, I’m not that acquianted with the code base to guarantee `mapped_ctrl_c` doesn’t have any other purpose, I can do some more digging and see if it’s used anywhere else but since we won’t be able to remove the condition that sets `ctrl_c_interrupts` (as it’s required for buffer ctrl-c mappings) the benefit of removing `mapped_ctrl_c` is not that impactful.

> how is anyone going to know the purpose of this in the future? git spelunking which leads to... a gigantic github discussion... is costly. We can easily compress the conclusion into a minimal comment.

Fair enough, this would be  my first commit to neovim, I am not yet familiar with comment preferences and best practices in this repo.